src/layouts/partial-panel-resolver.ts

```diff
@@ -184,6 +190,13 @@ class PartialPanelResolver extends HassRouterPage {
         this.appendChild(this._disconnectedPanel);
         this._disconnectedPanel = undefined;
       }
+      if (
+        this._disconnectedActiveElement &&
+        this._disconnectedActiveElement instanceof HTMLElement
+      ) {
+        this._disconnectedActiveElement.focus();
```

Comment:
In the future we should also add this logic to our router page for the cached elements.